apps/minifront/src/components/swap/swap-form/token-swap-input.tsx

```diff
@@ -38,66 +40,68 @@ const findMatchingBalance = (
 const isValidAmount = (amount: string, assetIn?: BalancesResponse) =>
   Number(amount) >= 0 && (!assetIn || !amountMoreThanBalance(assetIn, amount));
 
+const tokenSwapInputSelector = (state: AllSlices) => ({
+  swappableAssets: state.swap.swappableAssets,
+  assetIn: state.swap.assetIn,
+  setAssetIn: state.swap.setAssetIn,
+  assetOut: state.swap.assetOut,
+  setAssetOut: state.swap.setAssetOut,
+  amount: state.swap.amount,
+  setAmount: state.swap.setAmount,
+  balancesResponses: state.swap.balancesResponses,
+});
+
 /**
  * Exposes a UI with three interactive elements: an asset selector for the user
  * to choose which asset to swap _from_, an asset selector for the user to
  * choose which asset to swap _to_, and a text field for the user to enter an
  * amount.
  */
-export const TokenSwapInput = ({
-  assets = [],
-  balances = [],
-  amount,
-  onChangeAmount,
-  assetIn,
-  onChangeAssetIn,
-  assetOut,
-  onChangeAssetOut,
-  label,
-}: {
-  assets?: Metadata[];
-  balances?: BalancesResponse[];
-  amount: string;
-  onChangeAmount: (amount: string) => void;
-  assetIn?: BalancesResponse;
-  onChangeAssetIn: (assetIn: BalancesResponse) => void;
-  assetOut?: Metadata;
-  onChangeAssetOut: (assetOut: Metadata) => void;
-  label: string;
-}) => {
-  const balanceOfAssetOut = findMatchingBalance(assetOut, balances);
+export const TokenSwapInput = () => {
+  const {
+    swappableAssets,
+    amount,
+    setAmount,
+    assetIn,
+    setAssetIn,
+    assetOut,
+    setAssetOut,
+    balancesResponses,
+  } = useStoreShallow(tokenSwapInputSelector);
+  const balanceOfAssetOut = findMatchingBalance(assetOut, balancesResponses);
   const maxAmount = getAmount.optional()(assetIn);
   let maxAmountAsString: string | undefined;
   if (maxAmount) maxAmountAsString = joinLoHiAmount(maxAmount).toString();
 
   return (
-    <Box label={label}>
+    <Box label='Amount to swap'>
       <Input
         value={amount}
         type='number'
         inputMode='decimal'
         variant='transparent'
         placeholder='Enter an amount...'
         max={maxAmountAsString}
+        step='any'
```

Comment:
Fixes an issue from #1155 where Chrome expected the step size to be a certain number of decimal places